### PR TITLE
DDS Security API Reference [8430]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/326/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/329/)

File: docs/fastdds/api_reference/api_reference.rst [hunk 1]
```diff
@@ -14,3 +14,4 @@ Publish-Subscribe (RTPS) API that DDS implements for wired communications, as sp
 .. toctree::
 
    /fastdds/api_reference/includes/dds_pim/dds_pim
+   /fastdds/api_reference/dds_sec/dds_sec
```
Comment (EduPonz) [inline docs/fastdds/api_reference/api_reference.rst:17]: I'd call this directory `dds_security`. I believe it's more clear

File: docs/fastdds/api_reference/dds_sec/accesscontrol.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    /fastdds/api_reference/dds_sec/accesscontrol/accesscontrol
+    /fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributesmask
+    /fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributes
+    /fastdds/api_reference/dds_sec/accesscontrol/participantsecurityattributesmask
+    /fastdds/api_reference/dds_sec/accesscontrol/participantsecurityattributes
+    /fastdds/api_reference/dds_sec/accesscontrol/pluginendpointsecurityattributesmask
+    /fastdds/api_reference/dds_sec/accesscontrol/pluginendpointsecurityattributes
+    /fastdds/api_reference/dds_sec/accesscontrol/pluginparticipantsecurityattributesmask
+    /fastdds/api_reference/dds_sec/accesscontrol/pluginparticipantsecurityattributes
```
Comment (EduPonz) [inline docs/fastdds/api_reference/dds_sec/accesscontrol.rst:14]: Sort alphabetically.
```suggestion
    /fastdds/api_reference/dds_sec/accesscontrol/accesscontrol
    /fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributes
    /fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributesmask
    /fastdds/api_reference/dds_sec/accesscontrol/participantsecurityattributes
    /fastdds/api_reference/dds_sec/accesscontrol/participantsecurityattributesmask
    /fastdds/api_reference/dds_sec/accesscontrol/pluginendpointsecurityattributes
    /fastdds/api_reference/dds_sec/accesscontrol/pluginendpointsecurityattributesmask
    /fastdds/api_reference/dds_sec/accesscontrol/pluginparticipantsecurityattributes
    /fastdds/api_reference/dds_sec/accesscontrol/pluginparticipantsecurityattributesmask
```

File: docs/fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributes.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    :project: Fast RTPS
+
+.. doxygendefine:: ENDPOINT_SECURITY_ATTRIBUTES_FLAG_IS_KEY_PROTECTED
+    :project: Fast RTPS
+
+.. doxygendefine:: ENDPOINT_SECURITY_ATTRIBUTES_FLAG_IS_LIVELINESS_PROTECTED
+    :project: Fast RTPS
+
+.. doxygendefine:: ENDPOINT_SECURITY_ATTRIBUTES_FLAG_IS_VALID
+    :project: Fast RTPS
```
Comment (EduPonz) [inline docs/fastdds/api_reference/dds_sec/accesscontrol/endpointsecurityattributes.rst:34]: In general, sort alphabetically

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 4]
```diff
@@ -48,10 +58,14 @@ OwnershipQosPolicy
...
 parameterList
 params
+Pariticipant
+Partipants
 PartitionQosPolicy
 Pid
 pos
+preprocess
 PresentationQosPolicy
+prprocess
```
Comment (EduPonz) [inline docs/fastdds/api_reference/spelling_wordlist.txt:68]: I'd be good to correct this in Fast-RTPS

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 5]
```diff
@@ -95,4 +110,6 @@ TypeLookup
 TypeObjectV
 unregister
 untaken
+veryfies
+wether
```
Comment (EduPonz) [inline docs/fastdds/api_reference/spelling_wordlist.txt:114]: Same here

**Comment by EduPonz** [comment]
We have decided to close this, as the user will not interact with this API

**Pull request closed without merging**